### ISSUE TITLE
Datasource: Missing description on datasource API payload

### ISSUE THREAD

**Issue by franzwong** [grafana/grafana#71699]
### What happened?

Neither the web site [1] nor swagger UI explains the payload of datasource API. What I want to know are the possible values for "access" field.

I use the official grafana enterprise docker image and I can't find any description in Swagger UI too. It just describes that field is string.

I even check the source code and I can't see it [2].

Finally I just do a google search on grafana site and find it on "Provision Grafana" [3].

[1]: https://grafana.com/docs/grafana/latest/developers/http_api/data_source/#create-a-data-source

[2]: https://github.com/grafana/grafana/blob/release-10.0.2/pkg/services/datasources/models.go

[3]: https://grafana.com/docs/grafana/latest/administration/provisioning/#example-data-source-config-file

### What did you expect to happen?

I can find the description of payload on the web site. Just like dashboard API. [1]

[1]: [JSON Body schema:](https://grafana.com/docs/grafana/latest/developers/http_api/dashboard/#create--update-dashboard)

### Did this work before?

I don't know.

### How do we reproduce it?

1. Go to the web page
2. Read the page

### Environment (with versions)?

Grafana: 10.0.2
OS: N/A
Browser: N/A

### Grafana platform?

Docker

**Comment by mellieA** [comment]
@franzwong thank you for sharing your feedback, I'll get this request to the docs squad. 

To make it easier for our writers to visualize the changes, I'll paste the relevant bits of the docs here below, will you please confirm that I am asking for the right changes? 

- Add an API JSON Body Schema and describe values for the [datasource API payload docs](https://grafana.com/docs/grafana/latest/developers/http_api/data_source/#create-a-data-source) similar to what's documented for the [create/update dashboard endpoint](https://grafana.com/docs/grafana/latest/developers/http_api/dashboard/#create--update-dashboard): 

```
JSON Body schema:

dashboard – The complete dashboard model, id = null to create a new dashboard.
dashboard.id – id = null to create a new dashboard.
dashboard.uid – Optional unique identifier when creating a dashboard. uid = null will generate a new uid.
.
.
.
```
- Define the valid values for `access` in the datasource API. Currently, these parameters are documented in the provisioning docs, but only included in the [comments of a sample config file](https://grafana.com/docs/grafana/latest/administration/provisioning/#example-data-source-config-file): 

```
    # <string, required> Sets the access mode, either
    # proxy or direct (Server or Browser in the UI).
    # Some data sources are incompatible with any setting
    # but proxy (Server).
    access: proxy
```

**Comment by franzwong** [comment]
@mellieA That's correct. Also, it would be better if more explanation is given about proxy and direct.

**Comment by GrafanaWriter** [comment]
There are no resources to work on this from the technical writing team, @mellieA

**Comment by pp5311006** [comment]
Hi @papagian , I am interested in this issue, can you assign it to me?
I have created a draft PR, currently it's failing at build - enterprise step. I will look into it later today.

**Comment by johnsonchou3** [comment]
Hello if case no one is working on it, I would like to try it. 😃

**Comment by SplinterSword** [comment]
Hello is this issue still open
